### PR TITLE
added functional tests for mqtt demo #20

### PR DESCRIPTION
**Pull request by neelabalan**

### CODE REVIEW

File: tests/test_mqtt.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+from utils import MongoDb
+import paho.mqtt.client as mqtt
+
+influx_query_url = "http://localhost:8086/query?db=influx&"
+
+
+@pytest.fixture
+def collection():
+    con = MongoDb(host="localhost")
```
[the remark this answers is not in the document]
Comment (neelabalan) [reply tests/test_mqtt.py:16]: I've used the MongoDb class here

File: tests/test_mqtt.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        password = file.read()
+
+    mqtt_client = mqtt.Client()
+    mqtt_client.username_pw_set(username, password)
+    mqtt_client.connect("localhost", 1883)
+    yield mqtt_client
+    mqtt_client.disconnect()
+
+
+def publish_message(mqtt_client, topic, data):
```
[the remark this answers is not in the document]
Comment (neelabalan) [reply tests/test_mqtt.py:42]: sure

[the remark this answers is not in the document]
Comment (neelabalan) [reply tests/test_mqtt.py:42]: apart from the cleanup_influx I don't know where else to use type hinting. I couldn't find reference to same in mqtt.py, There seems to be no type hints in mqtt.py. Do you have any resources/link I can refer to?

File: tests/test_mqtt.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (danionescu0) [reply tests/test_mqtt.py:16]: it's ok, my bad

File: tests/test_mqtt.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (danionescu0) [reply tests/test_mqtt.py:42]: i've just made a commit, i've refactored mqtt.py to use objects and type hinting, take a look

[the remark this answers is not in the document]
Comment (neelabalan) [reply tests/test_mqtt.py:42]: This is helpful. Thank you. Will raise a new PR after making the change.

[the remark this answers is not in the document]
Comment (danionescu0) [reply tests/test_mqtt.py:42]: Great. You're welcome

**Comment by neelabalan** [comment]
are the new changes good?

**Comment by danionescu0** [comment]
The last pushed changes are mine: https://github.com/danionescu0/docker-flask-mongodb-example/commit/c93e09610b85ced6911696d1fe89611db1e2749e (this commit) Have you pushed them ?

**Comment by neelabalan** [comment]
> The last pushed changes are mine: [c93e096](https://github.com/danionescu0/docker-flask-mongodb-example/commit/c93e09610b85ced6911696d1fe89611db1e2749e) (this commit) Have you pushed them ?

yeah. I referred to this one for type hints. Didn't make changes on source file.

**Comment by neelabalan** [comment]
I synced up before adding test_mqtt.py

**Comment by danionescu0** [comment]
oh, yes you cand make a PR

**Comment by neelabalan** [comment]
> oh, yes you cand make a PR

did you mean I can't. If not I've already made the PR with the suggested changes

**Comment by danionescu0** [comment]
i've merged it

**Comment by neelabalan** [comment]
> i've merged it

I'll create a new PR and close previous one (if any changes suggested) going forward. It looks confusing.

**Pull request merged**